src/components/view/view-event-list.tsx

```diff
@@ -901,7 +921,29 @@ export class ViewEventList extends React.Component<ViewEventListProps> {
         // If we previously were scrolled to the bottom of the list, but now we're not,
         // scroll there again ourselves now.
         if (this.wasListAtBottom && !this.isListAtBottom()) {
-            this.listRef.current?.scrollToItem(this.props.events.length - 1);
+            this.listRef.current?.scrollToItem(this.props.events.length - 1);        
+        } else if (prevProps.selectedEvent !== this.props.selectedEvent && this.props.selectedEvent) {
+            // If the selected event changed and we have a selected event, scroll to it
+            // This handles restoring the selected event when returning to the tab
+            this.scrollToEvent(this.props.selectedEvent);
+        } else if (prevProps.filteredEvents.length !== this.props.filteredEvents.length) {
+            // If the filtered events changed (e.g., new events loaded), try to restore scroll position
+            setTimeout(() => {
+                this.restoreScrollPosition();
+            }, 50);
```

Comment:
I'm not sure I totally understand the goal of this bit. Can you explain when this applies in practice, and how it helps? I haven't noticed issues with the scroll position while filtering.